pkg/server/server.go

```diff
@@ -99,15 +102,21 @@ func (s *Service) Start(c context.Context) error {
 	return nil
 }
 
-func (s *Service) Stop(ctx context.Context) error {
-	if s.running.CompareAndSwap(true, false) {
-		return s.svr.Shutdown(ctx)
-	}
-
+func (s *Service) Stop() error {
 	if s.health != nil {
 		s.health.RemoveReadinessCheck(s.serverName())
 	}
 
+	if s.running.CompareAndSwap(true, false) {
+		ctx, cancel := context.WithTimeout(context.Background(), s.opts.ShutdownTimeout)
```

Comment:
should this not use the context given by the caller instead of context.Background()

Reply:
solved with https://github.com/sco1237896/sco-backend/pull/15/commits/f1d1b19430d57c43c073a48e5d090d562b7ff628